### DISCUSSION TITLE
How to understand the results from regexp.exec from multiple input path

### DISCUSSION THREAD

**Discussion by shitai-chen** [pillarjs/path-to-regexp#280] [General]
Hi Folks,

I was trying to do something like:
```
const regexp = pathToRegexp(["/v2/collections/:symbol/listings", "/v2/tokens/:token_id", "/v2/tokens/:token_id/listings"], keys);

regexp.exec("/v2/tokens/fooooooo/listings");
=> output: `/v2/tokens/btest/listings,,,btest`
```

I am trying to understand the response. My goal is to put all my api patterns into an array, then I can use your lib to detect which pattern the incoming URL falls into, however, I don't understand the output format. I don't know the array means. Why there are some empty fields there? and why 4 elements in the output array. 

Anyone can explain this to me? Super thanks.

Also I tried match function whose result format is more readable to me.  So I also wonder if I can set an array of strings to match function then match can tell me which one it matches. That would be great.

**Comment by blakeembrey** [comment]
It matches the `keys` array, just like when you use it without an array.
